### DISCUSSION TITLE
Understanding API library usage in checkstyle

### DISCUSSION THREAD

**Discussion by TheSuperEngineer** [checkstyle/checkstyle#12896] [General]
Identified at https://github.com/checkstyle/checkstyle/pull/12888#discussion_r1146032826 and https://github.com/checkstyle/checkstyle/pull/12888#discussion_r1146037195

#### Writing tests using the jupiter assertion library is prohibited in checkstyle codebase? Why? Rather Truth assertion library was emphasized.
(I think it is because code readability is increased in Truth testing library but I still need to be clear)

#### Is it just that jupiter assert api is prohibited to be used or the whole jupiter testing library is prohibited to be used and there is a requirement for the current jupiter based testing to migrate and use Truth testing library (image below shows jupiter testing library being used by checkstyle)

![image](https://user-images.githubusercontent.com/90137881/227358428-366012c2-414c-4808-bc5d-9b5e1b7bb109.png)

**Comment by rnveach** [comment]
> I think it is because code readability is increased in Truth testing library 

I believe it was because of the readability in the assertion failure messages, but it doesn't really matter. One form of writting tests is prohibited, because we want everything to be written in the library we have chosen. We want everything in the codebase going in a single direction and style.

> just that jupiter assert api is prohibited to be used or the whole jupiter testing library is prohibited 

I don't know the exact imports we support and don't. I do know we preferred the writing that is allowed with some throw assertions and that is why they are allowed. I assume our import control details what is allowed and what isn't. If you think something should be forbidden and isn't, I suggest making an issue for it.

**Reply by TheSuperEngineer** [reply]
> We want everything in the codebase going in a single direction and style.

yes. Too many testing libraries would make the testing difficult to handle and maintain in future. 
![image](https://user-images.githubusercontent.com/90137881/227368197-342c15a8-3d91-4eab-83eb-f419159ac40a.png)

**Reply by TheSuperEngineer** [reply]
Resolved.
